server/services/store/store.go

```diff
@@ -161,6 +163,15 @@ func IsErrNotFound(err error) bool {
 		return false
 	}
 
+	// check if this is a store.ErrNotFound
 	var nf *ErrNotFound
-	return errors.As(err, &nf)
+	if errors.As(err, &nf) {
+		return true
+	}
+
+	// check if this is a sql.ErrNotFound
+	if errors.Is(err, sql.ErrNoRows) {
```

Comment:
Note, an upcoming PR will standardize on `store.IsErrNotFound` for checking that condition instead of the mix of `store.ErrNotFound` and `sql.ErrNoRows`.